Broken TypeScript typings on fresh install

- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

After a fresh install of `@truffle/hdwallet-provider` (other packages possibly affected as well) TypeScript typings are not working correctly and compilation fails.

Mentioned in trufflesuite/ganache-core#465 but it's a different repo and didn't gain much traction

## Steps to Reproduce

1.  `npm install @truffle/hdwallet-provider web3`
2.  Put inside `src/index.ts`:
```js
import HDWalletProvider from "@truffle/hdwallet-provider";
console.log("Hello world!")
```
3.  `npx typescript --esModuleInterop src/index.ts`

## Expected Behavior

Compilation succeeds without any issues.

## Actual Results

Typecheck fails with the following errors:
```
node_modules/@truffle/hdwallet-provider/dist/index.d.ts:2:61 - error TS2307: Cannot find module 'ethereum-protocol'.
import { JSONRPCRequestPayload, JSONRPCErrorCallback } from "ethereum-protocol";
                                                             ~~~~~~~~~~~~~~~~~~~
node_modules/@truffle/hdwallet-provider/dist/index.d.ts:3:43 - error TS2307: Cannot find module 'web3/providers'.
import { Callback, JsonRPCResponse } from "web3/providers";
                                           ~~~~~~~~~~~~~~~~
```

## Environment

* Operating System: Tested on macOS and Arch Linux
* node version (`node --version`): Tested on `v10.18.0` and `v13.9.0`
* npm version (`npm --version`): `6.13.4`, `6.13.7`

--- a/packages/hdwallet-provider/package.json
+++ b/packages/hdwallet-provider/package.json
@@ -33,6 +33,7 @@
     "@types/ethereumjs-tx": "^1.0.1",
     "@types/ethereumjs-util": "^5.2.0",
     "@types/mocha": "^5.2.7",
+    "@types/web3": "^1.0.20",
     "@types/web3-provider-engine": "^14.0.0",
     "ganache-core": "2.10.2",
     "mocha": "6.2.0",

